pkg/printer/pod_printer.go

```diff
@@ -17,66 +17,71 @@ limitations under the License.
 package printer
 
 import (
-	"time"
+	"fmt"
 
-	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/utils/integer"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
 
 	cli "github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime"
-	"github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime/printer"
 	"github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime/printer/table"
 )
 
-const TerminatingPhase = "Terminating"
+func PodTablePrinter(c *cli.Config, tableResult runtime.Object) error {
+	tablePrinter := table.NewTablePrinter(table.PrintOptions{PaddingStart: paddingStart})
+	return tablePrinter.PrintObj(tableResult, c.Stdout)
+}
 
-func PodTablePrinter(c *cli.Config, podList *corev1.PodList) error {
-	printPodRow := func(pod *corev1.Pod, _ table.PrintOptions) ([]metav1beta1.TableRow, error) {
-		now := time.Now()
-		row := metav1beta1.TableRow{
-			Object: runtime.RawExtension{Object: pod},
+func DecodeIntoTable(info []*resource.Info) (runtime.Object, error) {
```

Comment:
Need unit tests for this code. Its doing the majority leg work of bridging kubectl lib output to runtime obj for CLI to consumes so we need to ensure this is covered in unit tests

Reply:
I have moved decode function along with resource builder and resource info details in to a separate function and have unit test cases to cover it all. We wouldn't need the Decode function public either